### PR TITLE
feat: Add merchant key store

### PR DESCRIPTION
**Pull request by dracarys18**
## Type of Change

- [x] Enhancement

## Description

This PR adds key store DB which stores encryption key per merchant account. The encryption key will be further encrypted by `master_key` which will further be KMS encrypted. 

### Additional Changes

- [x] This PR modifies the database schema
- [x] This PR modifies application configuration/environment variables

## How did you test it?

Manual

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code

### CODE REVIEW

File: crates/router/src/core/admin.rs [hunk 1]
```diff
@@ -100,6 +101,20 @@ pub async fn create_merchant_account(
             .attach_printable("Invalid routing algorithm given")?;
     }
 
+    let key_store = merchant_key_store::MerchantKeyStore {
+        id: None,
+        merchant_id: req.merchant_id.clone(),
+        key: crypto::Encryptable::encrypt(key.to_vec().into(), master_key, GcmAes256 {})
+            .await
+            .change_context(errors::ApiErrorResponse::InternalServerError)?,
```
Comment (jarnura) [inline crates/router/src/core/admin.rs:109]: Add attach printable for InternalServerError.

File: crates/router/src/services.rs [hunk 2]
```diff
@@ -177,3 +188,38 @@ impl Store {
...
+    #[cfg(not(feature = "kms"))]
+    let master_enc_key =
+        hex::decode(&conf.secrets.master_enc_key).expect("Failed to decode from hex");
+
+    master_enc_key
+}
+
+#[inline]
+pub fn generate_aes256_key(
+) -> crate::core::errors::CustomResult<[u8; 32], common_utils::errors::CryptoError> {
```
Comment (jarnura) [inline crates/router/src/services.rs:216]:
```suggestion
) -> errors::CustomResult<[u8; 32], common_utils::errors::CryptoError> {
```

**Review by jarnura — CHANGES_REQUESTED** [review]

**Review by NishantJoshi00 — APPROVED** [review]
LGTM

**Review by jarnura — COMMENTED** [review]
Add proper labels for the changes

File: config/config.example.toml [hunk 3]
```diff
@@ -79,6 +79,7 @@ sampling_rate = 0.1 # decimal rate between 0.0 - 1.0
 [secrets]
 admin_api_key = "test_admin" # admin API key for admin authentication
 jwt_secret = "secret"        # JWT secret used for user authentication
+master_enc_key = "73ad7bbbbc640c845a150f67d058b279849370cd2c1f3c67c4dd6c869213e13a" # Master Encryption key used to encrypt merchant wise encryption key
```
Comment (jarnura) [inline config/config.example.toml:82]: use a dummy key here, like `sample_key`

**Review by jarnura — CHANGES_REQUESTED** [review]
- All InternalServerError Should have attach_printable which explains the about the Error
- Import the mod and use mod::func in the application, change all functions which is imported as crate::mod::func to mod::func

**Comment by dracarys18** [comment]
> * All InternalServerError Should have attach_printable which explains the about the Error
> 
>     * Import the mod and use mod::func in the application, change all functions which is imported as crate::mod::func to mod::func

Done

**Review by jarnura — APPROVED** [review]

**Pull request merged**